apps/checkin/views.py

```diff
@@ -1,56 +1,163 @@
 
-    def get(self, request):
-        try:
-            application = Application.objects.get(user=request.user)
-        except Application.DoesNotExist:
-            return error_response(
-                "User does not have an application.",
-                (
-                    "Make sure that you are logged in with the account that "
-                    "created the application."
-                ),
-                404,
-            )
-        if not CONSTANTS.objects.get().DECISIONS_RELEASED:
-            return error_response(
-                "Decisions have not been released",
-                (
-                    "You can't check in until Totality has told you whether "
-                    "you've been admitted."
-                ),
-                403,
-            )
-        if application.admission_status != "A":
-            return error_response(
-                "User has not been admitted to Totality.",
-                "Current status is: {}".format(
-                    application.get_admission_status_display()),
-                403,
-            )
-        group = CheckInGroup.objects.get_or_create(applicant=request.user)[0]
-        group.save()
-        image = qrcode.make(group.id)
-        buffered = BytesIO()
-        image.save(buffered, format="JPEG")
-        return Response({"qrcode": base64.b64encode(buffered.getvalue())})
+    return success_data_jsonify({
+        "qr_code": group.id,
+        "name": application.first_name + " " + application.last_name,
+        "school": application.school,
+        "token": token,
+        "id": user.id
+    })
+
+
+@api_view(['GET'])
+def wallet(request):
+
+    headers = request.META
+    if "Authorization" not in headers:
+        return error_response("No authorization header provided.", 401)
+    token = headers["Authorization"].split("Bearer ")[1]
+    group_id = jwt.decode(token, SECRET_KEY)["group_id"]
+    try:
+        group = CheckInGroup.objects.get(id=group_id)
+    except CheckInGroup.DoesNotExist:
+        return error_response("Invalid jwt; no group exists with this id.",
+                              401)
+
+    user = group.applicant
+    cardInfo = EventTicket()
+    cardInfo.addPrimaryField('name', user.name, 'Name')
+    cardInfo.addHeaderField('header', 'October 12-14, 2018',
+                            'Brooklyn Expo Center')
+
+    if user.school:
+        cardInfo.addSecondaryField('loc', user.school, 'School')
+    cardInfo.addSecondaryField('email', user.username, 'Email')
+
+    organizationName = 'TotalityHacks'
+    passTypeIdentifier = 'pass.com.totalityhacks.totalityhacks'
+    teamIdentifier = ''  # TODO: Actually put this here.
+
+    passfile = Pass(cardInfo,
+                    passTypeIdentifier=passTypeIdentifier,
+                    organizationName=organizationName,
+                    teamIdentifier=teamIdentifier)
+
+    passfile.labelColor = 'rgb(255,255,255)'
+    passfile.foregroundColor = 'rgb(255,255,255)'
```

Comment:
Can you make a task on the design board to design the pass. For now assign it to me and then I'll re-assign to Ashley

Reply:
Done.